pkg/gatewayserver/io/basicstationlns/basicstationlns_util_test.go

```diff
@@ -16,42 +16,27 @@ package basicstationlns_test
 
 import (
 	"context"
-	"fmt"
 
 	"go.thethings.network/lorawan-stack/pkg/auth/rights"
 	"go.thethings.network/lorawan-stack/pkg/rpcmetadata"
 	"go.thethings.network/lorawan-stack/pkg/ttnpb"
 	"go.thethings.network/lorawan-stack/pkg/unique"
-	"google.golang.org/grpc/metadata"
 )
 
 func newContextWithRightsFetcher(ctx context.Context) context.Context {
 	return rights.NewContextWithFetcher(
 		ctx,
 		rights.FetcherFunc(func(ctx context.Context, ids ttnpb.Identifiers) (set *ttnpb.Rights, err error) {
 			uid := unique.ID(ctx, ids)
-			if uid != registeredGatewayUID {
+			if uid != registeredGatewayID.GatewayID {
```

Comment:
This can't be right; mixing UIDs with IDs. Introduce a `registeredGatewayUID` instead.